nixos/modules/services/web-apps/calibre-web.nix

```diff
@@ -0,0 +1,162 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.calibre-web;
+
+  inherit (lib) concatStringsSep mkEnableOption mkIf mkOption optional types;
+in
+{
+  options = {
+    services.calibre-web = {
+      enable = mkEnableOption "Calibre-Web";
+
+      listen = {
+        ip = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = ''
+            IP address that Calibre-Web should listen on.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 8083;
+          description = ''
+            Listen port for Calibre-Web.
+          '';
+        };
+      };
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "calibre-web";
+        description = ''
+          The directory below <filename>/var/lib</filename> where Calibre-Web stores its data.
+        '';
+      };
```

Comment:
Shouldn't the full path be configurable? Not that I would ever use anything outside of `/var/lib` anyway, but other modules allow for an arbitrary path in this case (e.g: `sonarr`).

Reply:
One can argue that it's easy to make a bind mount if necessary.

Reply:
> Shouldn't the full path be configurable?

I've looked through various modules and they use both approaches. Some use systemd `StateDirectory`, which implies that the directory is below `/var/lib`; some allow to configure full path. The benefit of `StateDirectory` is that systemd ensures that the directory exists and has correct user/group/permissions before starting the service. So I like it better.